1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/EnchantmentsDataGenerator.java

```diff
@@ -8,13 +8,18 @@
 import net.minecraft.registry.Registry;
 import net.minecraft.registry.RegistryKeys;
 import net.minecraft.registry.tag.TagKey;
+import net.minecraft.registry.entry.RegistryEntryList;
 import net.minecraft.util.Identifier;
+import net.minecraft.registry.entry.RegistryEntry;
+
 
 import java.util.List;
+import java.util.Set;
 
 public class EnchantmentsDataGenerator implements IDataGenerator {
-    public static String getEnchantmentTargetName(TagKey<Item> target) {
-        return target.id().getPath().split("/")[1];
+    public static String getEnchantmentTargetName(RegistryEntryList<Item> target) {
+        TagKey<Item> tagKey = target.getTagKey().orElseThrow();
+        return tagKey.toString();
```

Comment:
Replace with `return tagKey.id().getPath().split("/")[1];` , in this case toString() serializes the class name

Reply:
ah ok, i made a incorrect assumption based off of the 1.20.5 output that it was supposed to be outputted in the tagkey format